solc/CommandLineInterface.cpp

```diff
@@ -664,31 +664,39 @@ bool CommandLineInterface::parseLibraryOption(string const& _input)
 	for (string const& lib: libraries)
 		if (!lib.empty())
 		{
-			//search for last colon in string as our binaries output placeholders in the form of file:Name
-			//so we need to search for the second `:` in the string
-			auto colon = lib.rfind(':');
-			if (colon == string::npos)
+			//search for equal sign in string as our binaries output placeholders in the form of file=Name
+			//so we need to search for `=` in the string
+			solAssert(lib.rfind('=') == lib.find('='), "Only one equal sign \"=\" is allowed in the address string \"" + lib + "\".");
+			auto equalSign = lib.find('=');
+			if (equalSign == string::npos)
 			{
-				serr() << "Colon separator missing in library address specifier \"" << lib << "\"" << endl;
+				serr() << "Equal sign separator missing in library address specifier \"" << lib << "\"" << endl;
 				return false;
 			}
 
-			string libName(lib.begin(), lib.begin() + static_cast<ptrdiff_t>(colon));
+			string libName(lib.begin(), lib.begin() + static_cast<ptrdiff_t>(equalSign));
 			boost::trim(libName);
 			if (m_libraries.count(libName))
 			{
 				serr() << "Address specified more than once for library \"" << libName << "\"." << endl;
 				return false;
 			}
 
-			string addrString(lib.begin() + static_cast<ptrdiff_t>(colon) + 1, lib.end());
+			string addrString(lib.begin() + static_cast<ptrdiff_t>(equalSign) + 1, lib.end());
 			boost::trim(addrString);
-			if (addrString.substr(0, 2) == "0x")
+			if (addrString.substr(0, 2) == "0x" || addrString.substr(0, 2) == "0X")
```

Comment:
Is `0X` actually used?

Reply:
I will remove this part for ```0X``` if solidity CLI does not allow the prefix.